### PR TITLE
CLI refactor

### LINKED ISSUE #1280: Convoy CLI Refined
We're enabling local development on Incoming projects.

### PR DESCRIPTION
**Pull request by danvixent**
partly fixes #1280

### CODE REVIEW

File: internal/pkg/socket/handlers.go [hunk 1]
```diff
@@ -82,9 +82,9 @@ func ListenHandler(hub *Hub, repo *Repo) http.HandlerFunc {
 		}
 
 		project := m.GetProjectFromContext(r.Context())
-		app := m.GetEndpointFromContext(r.Context())
+		// app := m.GetEndpointFromContext(r.Context())
```
Comment (jirevwe) [inline internal/pkg/socket/handlers.go:85]: If this isn't need why not just remove it

File: internal/pkg/socket/handlers.go [hunk 2]
```diff
@@ -111,15 +111,22 @@ func LoginHandler(hub *Hub, repo *Repo) http.HandlerFunc {
 		}
 
 		project := m.GetProjectFromContext(r.Context())
-		endpoint := m.GetEndpointFromContext(r.Context())
+		// endpoint := m.GetEndpointFromContext(r.Context())
```
Comment (jirevwe) [inline internal/pkg/socket/handlers.go:114]: Same with this one and the other commented out lines of code

**Review by jirevwe — APPROVED** [review]

File: internal/pkg/middleware/middleware.go [hunk 3]
```diff
@@ -574,6 +574,22 @@ func (m *Middleware) GetDefaultProject(r *http.Request, projectRepo datastore.Pr
 	return projects[0], err
 }
 
+func (m *Middleware) RequirePersonalAccessToken() func(next http.Handler) http.Handler {
```
Comment (Subomi) [inline internal/pkg/middleware/middleware.go:577]: @danvixent JWTs are authenticated through the native realm. Will it go through this?

**Review by Subomi — CHANGES_REQUESTED** [review]
@danvixent FYI

File: internal/pkg/middleware/middleware.go (hunk 3 above)
Comment (danvixent) [reply internal/pkg/middleware/middleware.go:577]: @Subomi it won't, we check the AuthenticatedBy field.

File: internal/pkg/socket/client.go [hunk 4]
```diff
@@ -169,14 +168,14 @@ func (c *Client) IsOnline() bool {
...
 
-func (c *Client) HasEventType(evType string) bool {
-	for _, eventType := range c.EventTypes {
-		if evType == eventType || eventType == "*" {
-			return true
-		}
-	}
-	return false
-}
+//func (c *Client) HasEventType(evType string) bool {
```
[the remark this answers is not in the document]
Comment (danvixent) [reply internal/pkg/socket/client.go:171]: @Subomi It's not being used though.

File: cmd/stream.go [hunk 5]
```diff
@@ -89,7 +90,7 @@ func addStreamCommand(a *app) *cobra.Command {
 
 			srv.SetHandler(handler)
 
-			a.logger.Infof("Stream server running on port %v", socketPort)
+			log.Infof("Stream server running on port %v", socketPort)
```
[the remark this answers is not in the document]
Comment (danvixent) [reply cmd/stream.go]: fixed

File: internal/pkg/middleware/middleware.go (hunk 3 above)
Comment (danvixent) [reply internal/pkg/middleware/middleware.go:577]: @Subomi they are actually authenticated by the JWT realm

Comment (Subomi) [reply internal/pkg/middleware/middleware.go:577]: Make sense. @danvixent 

How can we ensure it's only PATs that can access this route, not all API Keys authenticated by the native realm.

File: internal/pkg/socket/client.go (hunk 4 above)
[the remark this answers is not in the document]
Comment (Subomi) [reply internal/pkg/socket/client.go:171]: @danvixent Then we should remove it entirely.

File: worker/task/process_event_creation.go [hunk 6]
```diff
@@ -227,16 +237,24 @@ func findSubscriptions(ctx context.Context, endpointRepo datastore.EndpointRepos
...
+			log.WithError(err).Error("error fetching subscriptions for this source")
 			return subscriptions, &EndpointError{Err: errors.New("error fetching subscriptions for this source"), delay: 10 * time.Second}
 		}
 
 		subscriptions, err = matchSubscriptionsUsingFilter(ctx, event, subRepo, subs)
 		if err != nil {
 			return subscriptions, &EndpointError{Err: errors.New("error fetching subscriptions for this source"), delay: 10 * time.Second}
 		}
+
+		cliSubs, err := subRepo.FindCLISubscriptions(ctx, project.UID)
```
Comment (Subomi) [inline worker/task/process_event_creation.go:251]: @danvixent Is there a reason we are doing this? My assumption is `FindSubscriptionsBySourceID` should return all `subs` both for endpoints and devices, and then all subs with matching filters are returned.

**Review by Subomi — APPROVED** [review]

**Pull request merged**